### PR TITLE
redundant code in tutorial; allowing bool to be passed in dropout

### PR DESCRIPTION
**Pull request by seanccho**

### CODE REVIEW

**Review by pavithrasv — APPROVED** [review]

**Review by pavithrasv — COMMENTED** [review]
Seems to me that the doc string for Model needs to be updated. The following lines and the code do not match

```
  If you subclass `Model`, you can optionally have	  If you subclass `Model`, you can optionally have
  a `training` argument (boolean) in `call`, which you can use to specify	  a `training` argument (boolean) in `call`, which you can use to specify
  a different behavior in training and inference:	  a different behavior in training and inference:
```

**Comment by pavithrasv** [comment]
> Seems to me that the doc string for Model needs to be updated. The following lines and the code do not match
> 
> ```
>   If you subclass `Model`, you can optionally have	  If you subclass `Model`, you can optionally have
>   a `training` argument (boolean) in `call`, which you can use to specify	  a `training` argument (boolean) in `call`, which you can use to specify
>   a different behavior in training and inference:	  a different behavior in training and inference:
> ```

Discussed with @omalleyt12 who worked on training arg related changes recently. Having training arg in this use case is still valid as users can explicitly pass training=False. Would like to keep the doc in the file as is.

**Comment by seanccho** [comment]
@pavithrasv @omalleyt12 
Right, the users can pass training=False. In fact, the transformer example (https://www.tensorflow.org/beta/tutorials/text/transformer#encoder_and_decoder) is passing training=False as a simple test (`sample_encoder_layer_output = sample_encoder_layer(
    tf.random.uniform((64, 43, 512)), False, None)` this is in the second line of the second code snippet under https://www.tensorflow.org/beta/tutorials/text/transformer#encoder_and_decoder).

Yet the dropout layer (tensorflow/python/keras/layers/core.py line 154) is only checking if training is None. If we change that to if not training, that call can be used for both the None/not None case and the True/False case.

**Comment by omalleyt12** [comment]
> @pavithrasv @omalleyt12
> Right, the users can pass training=False. In fact, the transformer example (https://www.tensorflow.org/beta/tutorials/text/transformer#encoder_and_decoder) is passing training=False as a simple test (`sample_encoder_layer_output = sample_encoder_layer( tf.random.uniform((64, 43, 512)), False, None)` this is in the second line of the second code snippet under https://www.tensorflow.org/beta/tutorials/text/transformer#encoder_and_decoder).
> 
> Yet the dropout layer (tensorflow/python/keras/layers/core.py line 154) is only checking if training is None. If we change that to if not training, that call can be used for both the None/not None case and the True/False case.

@seanccho The `training is None` check is done to substitute in the learning_phase() Tensor. If a user passes in `False` explicitly, this substitution should not be done. Also, symbolic Tensors can not be part of a boolean expression, and since `training` can be a Tensor, this has the potential to throw errors

Example (assuming 1.x):

```
import tensorflow as tf
x = tf.ones((10, 10))
if x:  # will error out
  print('hey')
```

**Comment by seanccho** [comment]
Thanks for the clarification!
```
import tensorflow as tf
x = tf.constant(True) # T/F flag
if x:
  print('hey')
```
yields a TypeError which says to Use `if t is not None:` instead of `if t:` in 1.14.
